File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mr/nu/chidrenass/d/ChidrenassAggDO.java
package iih.ci.mr.nu.chidrenass.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;
import xap.mw.coreitf.i.IBaseDO;
import iih.ci.mr.nu.chidrenass.d.desc.ChidrenassAggDODesc;



/**
 * 高危儿观察护理记录单
 */
public class ChidrenassAggDO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public ChildrenInAsseDO getParentDO() {
		return ((ChildrenInAsseDO) super.getParentDO());
	}

	public void setParentDO(ChildrenInAsseDO headDO) {
		setParent(headDO);
	}

	public ChildrenInAsseNurRecordDO[] getChildrenInAsseNurRecordDO() {
		IBaseDO[] dos = getChildren(ChildrenInAsseNurRecordDO.class);
		if(dos==null) return null;
		ChildrenInAsseNurRecordDO[] result=new ChildrenInAsseNurRecordDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(ChildrenInAsseNurRecordDO)dos[i];
		}
		return result;
	}
	
	public void setChildrenInAsseNurRecordDO(ChildrenInAsseNurRecordDO[] dos) {
		setChildren(ChildrenInAsseNurRecordDO.class, dos);
	}

	@Override
	public IAggDesc getAggDesc() {
		ChidrenassAggDODesc desc = new ChidrenassAggDODesc();

		return desc;
	}
	
	 public  BaseDO createParentDO() {
         return new ChildrenInAsseDO();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.mr.nu.chidrenass.d.ChildrenInAsseNurRecordDO")) {
             return new ChildrenInAsseNurRecordDO();
         }
         return null;
     }
}

File: client/iih_stub/iih.ci_stub/iih.ci.mr/nu/chidrenass/d/ChidrenassAggDO.cs
using System;
using xap.mw.core.data;
using xap.mw.core.meta;

namespace iih.ci.mr.nu.chidrenass.d
{
	/// <summary>
	/// 高危儿观察护理记录单DO
	/// </summary>
	public class ChidrenassAggDO : BaseAggDO
	{	
	    private static string CHILDRENINASSEDO_CHILDRENINASSENURRECORDDO="iih.ci.mr.nu.chidrenass.d.ChildrenInAsseNurRecordDO";

        public ChidrenassAggDO() {
            this.setParentDO(new ChildrenInAsseDO());
        }

  	    public ChildrenInAsseDO getParentDO() {
		    return ((ChildrenInAsseDO) base.ParentDO);
	    }      
	    
	    public void setParentDO(ChildrenInAsseDO headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public ChildrenInAsseNurRecordDO[] getChildrenInAsseNurRecordDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(CHILDRENINASSEDO_CHILDRENINASSENURRECORDDO));
		    if (dos == null || dos.Length==0){return null;}
            ChildrenInAsseNurRecordDO[] r = new ChildrenInAsseNurRecordDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as ChildrenInAsseNurRecordDO;
            }
	        return r;
		    
	    }
	    
	    public void setChildrenInAsseNurRecordDO(ChildrenInAsseNurRecordDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new ChildrenInAsseDO();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (CHILDRENINASSEDO_CHILDRENINASSENURRECORDDO == clzName)
	        {
                return new ChildrenInAsseNurRecordDO();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.mr.nu.chidrenass.d.ChildrenInAsseNurRecordDO"};
        }
	    public new static AggMeta GetMeta()
	    {
	        AggMeta meta = new AggMeta();
	        meta.OneToOneDOs = new string[]{};
	        return meta;
	    }
	    
	}
}
